### DISCUSSION TITLE
Loss Questions

### DISCUSSION THREAD

**Discussion by Hadri59** [MilesCranmer/PySR#311] [Q&A] [answered]
Hello, 
I was wondering if there is a way to compute a loss function that is indicative of linearity. 
I am trying to find an equation that is not purely equal to y, but proportional. I don't not mind if there is a coeff or an intercept. 
How to extract a metric that is indicative of linearity, like the total sum of squared, and use it as the loss function (in using the python API)

Thanks :)

**Comment by MilesCranmer** [comment]
I think there isn't a way to write an elementwise loss function that gives you this. However, I think you could use the branch in #143, and use some measure of correlation for your loss.

For example, the Pearson correlation coefficient: https://en.wikipedia.org/wiki/Pearson_correlation_coefficient.
![image](https://user-images.githubusercontent.com/7593028/205958261-282d8917-820f-4147-bf0a-6f634b922dc6.png)

You could take the absolute value of it as your loss function. You would need to specify the full loss function, rather than just the elementwise loss.

In other words, take this function: https://docs.julialang.org/en/v1/stdlib/Statistics/#Statistics.cor and essentially replace this line:
https://github.com/MilesCranmer/SymbolicRegression.jl/blob/9034341558382188b6c4a9092d847ca8a7162b1d/src/LossFunctions.jl#L45
with it.

**Comment by Hadri59** [comment]
Alright ! Thanks a lot for the advice ! I will take a look :)

**Comment by MilesCranmer** [comment, marked as answer]
Updated answer is to pass a custom loss function to `full_objective` (PySR) or `loss_function` (SymbolicRegression.jl).
